python/ray/experimental/slib/operator.py

```diff
@@ -0,0 +1,77 @@
+import enum
+
+# Stream partitioning schemes
+class PScheme(object):
+    def __init__(self,strategy,partition_fn=None):
+        self.strategy = strategy
+        self.partition_fn = partition_fn
+
+    def __repr__(self):
+        return "({},{})".format(self.strategy,self.partition_fn)
+
+# Partitioning strategies
+class PStrategy(enum.Enum):
+    Forward = 0     # Default
+    Shuffle = 1
+    Rescale = 2
+    RoundRobin = 3
+    Broadcast = 4
+    Custom = 5
+    ShuffleByKey = 6
+    # ...
+
+# Operator types
+class OpType(enum.Enum):
+    Source = 0
+    Map = 1
+    FlatMap = 2
+    Filter = 3
+    TimeWindow = 4
+    KeyBy = 5
+    Sink = 6
+    WindowJoin = 7
+    Inspect = 8
+    ReadTextFile = 9
+    Reduce = 10
+    Sum = 11
+    # ...
+
+# A logical dataflow operator
+class Operator(object):
+    def __init__(self,
+                 id,
+                 type,
```

Comment:
`id` and `type` are both built in Python functions so ideally choose different names